src/components/LayoutTree.tsx

```diff
@@ -364,16 +370,27 @@ const LayoutTree = () => {
         state.cursor.length - (hasChildren(state, head(state.cursor)) ? 2 : 3)
       : 0,
   )
+  const scrollTop = scrollTopStore.useState()
 
-  const [bulletWidth, setBulletWidth] = useState<number | undefined>(0)
+  // Width of thought bullet
+  const [bulletWidth, setBulletWidth] = useState(0)
+  // Height of toolbar element
+  const [toolbarHeight, setToolbarHeight] = useState(0)
+  // Distance from toolbar to the first visible thought
+  const [distanceFromTop, setDistanceFromTop] = useState(0)
 
   // set the bullet width only during drag or when simulateDrop is true
   useLayoutEffect(() => {
     if (dragInProgress || testFlags.simulateDrop) {
       const bullet = ref.current?.querySelector('[aria-label=bullet]')
-      setBulletWidth(bullet?.getBoundingClientRect().width)
+      if (bullet) setBulletWidth(bullet?.getBoundingClientRect().width)
+
+      const toolbar = document.querySelector('#toolbar')
+      if (toolbar) setToolbarHeight(toolbar.getBoundingClientRect().height)
+
+      setDistanceFromTop((ref.current?.getBoundingClientRect().top ?? 0) + scrollTop)
```

Comment:
You add `scrollTop` here only to subtract it [below](https://github.com/cybersemics/em/pull/2369/files#diff-ed74b196536d8a76f93f0d172796b6356ed7b7d3775bc7840ce7b15648dfcfb2R477). i.e. I think it cancels out when you calculate `maxVisibleY`. Therefore, you can remove `scrollTop` from these calculations and avoid having to trigger `useLayoutEffect` on scroll.

Reply:
f7404afb9a